widget/task_preview.lua

```diff
@@ -94,23 +75,23 @@ local enable = function(opts)
     local widget_border_width = beautiful.task_preview_widget_border_width or
                                     dpi(3)
 
-    local task_preview_box = wibox({
-        type = "dropdown_menu",
-        visible = false,
-        ontop = true,
-        input_passthrough = true,
-        width = widget_width,
-        height = widget_height,
-        bg = "#00000000"
-    })
+    local task_preview_box = nil
 
     awesome.connect_signal("bling::task_preview::visibility", function(s, v, c)
-        draw_widget(c, task_preview_box, screen_radius, widget_bg,
-                    widget_border_color, widget_border_width, margin)
+        if v then 
+            task_preview_box = awful.popup({
```

Comment:
The problem with creating the popup on every signal call is that it would be spawned first at the origin of the screen (top left) and then slightly move to the position stated by the placement function or x/y coordinates. This was kinda noticeable when I was migrating the tag preview to popups. The solution is to create the popup outside the function with an empty background container as the `widget` property like [here](https://github.com/BlingCorp/bling/blob/37edd6e1eb257891e7cd461405365ae19e5f8832/widget/tag_preview.lua#L151), and update the `widget` property with the `setup` function like [here](https://github.com/BlingCorp/bling/blob/37edd6e1eb257891e7cd461405365ae19e5f8832/widget/tag_preview.lua#L175).

Reply:
I remember noticing this and trying to have the popup be created before but I never thought of just using a dummy widget

Reply:
Yup! You can always take a look at the tag preview widget on how its done, like I said before I would want the way we do the widgets to be consistent :smile: